hw/xbox/mcpx_apu.c

```diff
@@ -1447,17 +1477,21 @@ static void process_voice(MCPXAPUState *d,
       voice_set_mask(d, v, NV_PAVS_VOICE_PAR_OFFSET, NV_PAVS_VOICE_PAR_OFFSET_CBO, cbo);
     }
 
+#if 0
     // Apply the amplitude envelope
     //FIXME: Figure out when exactly and how exactly this is actually done
     for(unsigned int j = 0; j < channels; j++) {
         for(unsigned int i = 0; i < 0x20; i++) {
             samples[j][i] *= ea_value;
         }
     }
+#endif
```

Comment:
Why is this being disabled? While the exact place isn't known (especially for feedback-loops, 3D audio, effects, ..) the code itself should be fine.

If you want to disable it, you can simply stub the envelope.

Reply:
If I recall correctly, there seemed to be an issue with this amplitude scaling. We should probably understand this better before we apply it to the samples. Agree we can stub in envelope generator.